### PR TITLE
Repeater features

### PR DESCRIPTION
**Pull request by benrudolph**
This was my hackathon project. Adds a new report and some more features for repeaters. Taking screenshots and adding details below:

On repeater list page, allows you to see "Success", "Pending", and "Failed" for each repeater and links you to a RepeatRecord report
screen shot 2016-02-14 at 2 33 34 pm

The report allows you to view records and their current state. It allows you to see the payload and then resend the payload if you wish.
screen shot 2016-02-14 at 2 34 39 pm
screen shot 2016-02-14 at 2 34 51 pm
screen shot 2016-02-14 at 2 34 45 pm

Adding label cause i want to do a thorough review myself before merge. not exactly sure who to ping, but i know @czue has been doing some debug work on repeaters and @dannyroberts i think was the original author

cc: @gcapalbo

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 6 errors:

* corehq/apps/domain/views.py, line 35 - F811 redefinition of unused 'use_jquery_ui' from line 35
* corehq/apps/domain/views.py, line 35 - F811 redefinition of unused 'use_select2' from line 35
* corehq/apps/repeaters/dbaccessors.py, line 37 - E124 closing bracket does not match visual indentation
* corehq/apps/repeaters/dbaccessors.py, line 62 - E124 closing bracket does not match visual indentation
* corehq/apps/repeaters/views.py, line 1 - F401 'json' imported but unused
* corehq/couchapps/receiverwrapper/views/repeat_records/map.js, line 1 - Missing name in function declaration.

File: corehq/apps/repeaters/dbaccessors.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        ids = sorted(_get_repeater_ids_by_domain(domain))
+        startkey = [domain, ids[0], state]
+        endkey = [domain, ids[-1], state]
+
+    result = RepeatRecord.get_db().view('receiverwrapper/repeat_records',
+        startkey=startkey,
+        endkey=endkey,
+        include_docs=False,
+        reduce=True,
+    ).one()
```
Comment (dimagimon) [inline corehq/apps/repeaters/dbaccessors.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/repeaters/dbaccessors.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        kwargs['endkey'] = [domain, repeater_id, {}]
+    elif repeater_id and state:
+        kwargs['startkey'] = [domain, repeater_id, state]
+        kwargs['endkey'] = [domain, repeater_id, state]
+    elif not repeater_id and state:
+        kwargs['key'] = [domain, None, state]
+
+    results = RepeatRecord.get_db().view('receiverwrapper/repeat_records',
+        **kwargs
+    ).all()
```
Comment (dimagimon) [inline corehq/apps/repeaters/dbaccessors.py]: E124 closing bracket does not match visual indentation

File: corehq/couchapps/receiverwrapper/views/repeat_records/map.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+function (doc) {
```
Comment (dimagimon) [inline corehq/couchapps/receiverwrapper/views/repeat_records/map.js]: Missing name in function declaration.

**Comment by czue** [comment]
very cool! looks super useful and I think will save support a bunch of time. happy to review when you've had a chance to review on your own.

File: corehq/apps/repeaters/tests/test_dbaccessors.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
+from mock import patch
```
Comment (dimagimon) [inline corehq/apps/repeaters/tests/test_dbaccessors.py]: F401 'patch' imported but unused

**Comment by snopoke** [comment]
This is great. I've wanted these features every time I work with repeaters! One outstanding feature: being able to edit an existing repeater.

**Comment by benrudolph** [comment]
that was actually on the list but i didn't get time to do it haha

**Comment by benrudolph** [comment]
ok, had time to review this myself and i think it's ready for review

**Comment by gcapalbo** [comment]
nice work - looks good to me, just the one comment about the celery queue. will give @czue or @dannyroberts a chance to review as well.

also, added a reindex label to this as it adds a new view

File: corehq/apps/domain/templates/domain/repeat_record_report.html [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                      }
+
+                  })
+                  .fail(function(data) {
+                      $btn.removeSpinnerFromButton();
+                      $btn.text(gettext('Failed to send'));
+                      $btn.addClass('btn-danger');
+                  });
+          });
+    });
```
Comment (snopoke) [inline corehq/apps/domain/templates/domain/repeat_record_report.html]: does this conform to our JS best practices (if we have them yet)? @dimagi/js-team

File: corehq/apps/domain/views.py [hunk 6]
```diff
@@ -2069,6 +2086,119 @@ def friendly_repeater_names(self):
...
+    def _make_view_payload_button(self, record_id):
+        return '''
+        <a
+            class="btn btn-default"
+            role="button"
+            data-record-id={}
+            data-toggle="modal"
+            data-target="#view-record-payload-modal">
+            View Payload
+        </a>
```
Comment (snopoke) [inline corehq/apps/domain/views.py]: I know there probably isn't a way to get around this now but ug, it smells!

**Comment by snopoke** [comment]
Looks good! :boot: :construction:

**Comment by czue** [comment]
looks great and also nice tests and cleanup! not merging in case we're worried about reindexing (though guessing is fine because it's the `receiverwrapper` db)

File: corehq/apps/domain/views.py (hunk 6 above)
Comment (benrudolph) [inline corehq/apps/domain/views.py]: sadly we return rendered HTML to the report so there is no way around this unless we do a major refactor :disappointed:

File: corehq/apps/domain/templates/domain/repeat_record_report.html (hunk 5 above)
Comment (benrudolph) [inline corehq/apps/domain/templates/domain/repeat_record_report.html]: ouf realize some of the formatting is whack (used an automatic formatter to fix some of the html below). i think for this small case it's ok for this to be inline and not using a library as it's very basic dom manipulation; open to push back though

**Comment by benrudolph** [comment]
don't think the reindex is a big issue as you mentioned

**Comment by benrudolph** [comment]
:white_check_mark:

**Pull request merged**